src/nvim/drawline.c

```diff
@@ -1551,6 +1551,11 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool number_onl
       }
       // don't clear anything after wlv.col
       win_put_linebuf(wp, wlv.row, 0, wlv.col, wlv.col, bg_attr, false);
+      // update the 'statuscolumn' for the entire line size
+      if (number_only && statuscol.draw
+          && ++wlv.row - wlv.startrow < wp->w_lines[wlv.lnum - wp->w_topline].wl_size) {
+        continue;
```

Comment:
Might need to reset some winlinevars (wlv->off?) before continuing, I'll test later.

Reply:
> Might need to reset some winlinevars (wlv->off?) before continuing, I'll test later.

That and set `draw_cols` to true to make sure the 'statuscolumn' is actually being re-evaluated, instead of just duplicated. Good to go now I think.